tests/e2e/stateapp/stateapp_test.go

```diff
@@ -608,38 +608,58 @@ func TestQueryStateStore(t *testing.T) {
 	_, err := utils.HTTPGetNTimes(externalURL, numHealthChecks)
 	require.NoError(t, err)
 
-	// Populate store.
-	url := fmt.Sprintf("%s/test/%s/%s/querystatestore", externalURL, "http", "load")
-	body, err := os.ReadFile("query-data/dataset.json")
-	require.NoError(t, err)
-
-	_, status, err := utils.HTTPPostWithStatus(url, body)
-	require.NoError(t, err)
-	require.Equal(t, 204, status)
-
-	tests := []struct {
-		path string
-		keys int
+	for _, store := range []struct {
+		name    string
+		isRedis bool
 	}{
 		{
-			path: "./query-data/query.json",
-			keys: 3,
+			name:    "querystatestore",
+			isRedis: false,
 		},
-	}
-	for _, test := range tests {
-		body, err := os.ReadFile(test.path)
+		{
+			name:    "redisquerystore",
+			isRedis: true,
+		},
+	} {
+		// Populate store.
+		url := fmt.Sprintf("%s/test/%s/%s/%s", externalURL, "http", "load", store.name)
+		if store.isRedis {
+			url += "?contentType=application/json"
+		}
+		body, err := os.ReadFile("query-data/dataset.json")
 		require.NoError(t, err)
 
-		for _, protocol := range []string{"http", "grpc"} {
-			url := fmt.Sprintf("%s/test/%s/%s/querystatestore", externalURL, protocol, "query")
-			resp, status, err := utils.HTTPPostWithStatus(url, body)
-			require.NoError(t, err)
-			require.Equal(t, 200, status)
+		_, status, err := utils.HTTPPostWithStatus(url, body)
+		require.NoError(t, err)
+		require.Equal(t, 204, status)
 
-			var states requestResponse
-			err = json.Unmarshal(resp, &states)
+		tests := []struct {
+			path string
+			keys int
+		}{
+			{
+				path: "./query-data/query.json",
+				keys: 3,
+			},
+		}
+		for _, test := range tests {
+			body, err := os.ReadFile(test.path)
 			require.NoError(t, err)
-			require.Equal(t, test.keys, len(states.States))
+
+			for _, protocol := range []string{"http", "grpc"} {
+				url := fmt.Sprintf("%s/test/%s/query/%s", externalURL, protocol, store.name)
+				if store.isRedis {
+					url += "?contentType=application/json&queryIndexName=orgIndx"
```

Comment:
I think it is OK to add set the queryIndexName here. Can these two be moved to the component metadata, so we don't need this to be in the request? The fact that our E2E test now must know about being redis or not is a sign that we are breaking portability.